### PR TITLE
Add EditRoute flag to RouteSRV

### PR DESCRIPTION
**Pull request by MustafaSaber**
Follow up on https://github.bus.zalan.do/pg9/issues/issues/33

Right now CloneRoute testcase fail but we really care now about EditRoute. So to follow up on CloneRoute check here https://github.com/zalando/skipper/pull/2132

Signed-off-by: Mustafa Abdelrahman <mustafa.abdelrahman@zalando.de>

### CODE REVIEW

File: routesrv/polling.go [hunk 1]
```diff
@@ -79,6 +81,16 @@ func (p *poller) poll(wg *sync.WaitGroup) {
 		if p.oauth2Config != nil {
 			routes = p.oauth2Config.NewGrantPreprocessor().Do(routes)
 		}
+		if p.editRoute != nil {
```
Comment (AlexanderYastrebov) [inline routesrv/polling.go:84]: Go allows range over nil slice https://go.dev/play/p/wJ-Rcpo2jnl

File: routesrv/polling.go [hunk 2]
```diff
@@ -79,6 +81,16 @@ func (p *poller) poll(wg *sync.WaitGroup) {
...
+			for _, editor := range p.editRoute {
+				routes = editor.Do(routes)
+			}
+		}
+		if p.cloneRoute != nil {
+			for _, cloner := range p.cloneRoute {
+				routes = cloner.Do(routes)
+			}
+		}
```
Comment (AlexanderYastrebov) [inline routesrv/polling.go:93]: Route processing starts to grow, lets have a helper `process` or `prepare` method  and move all preprocessors as well as `sort.SliceStable` (from below) there.

Comment (AlexanderYastrebov) [inline routesrv/polling.go:93]: Route processing starts to grow, lets have a helper `process` or `prepare` method  and move all preprocessors as well as `sort.SliceStable` (from below) there.

Comment (szuecs) [reply routesrv/polling.go:93]: what?

File: routesrv/polling.go [hunk 3]
```diff
@@ -127,3 +126,22 @@ func (p *poller) poll(wg *sync.WaitGroup) {
...
+	}
+	for _, editor := range p.editRoute {
+		routes = editor.Do(routes)
+	}
+
+	for _, cloner := range p.cloneRoute {
+		routes = cloner.Do(routes)
+	}
+
+	return routes
```
Comment (AlexanderYastrebov) [inline routesrv/polling.go:146]: lets also move https://github.com/zalando/skipper/blob/566fe8c4ad7f7d46da7a2bb9b8a1f6bb75125457/routesrv/polling.go#L102-L105 here (no need to check length, sort would be no-op for empty slice)

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by MustafaSaber** [comment]
👍

**Pull request merged**